src/Obsidian/Cache.ts

```diff
@@ -355,10 +355,42 @@ export class Cache {
                 }
             }
         }
-
+        this.linkRelations(tasks);
         return tasks;
     }
 
+    private linkRelations(tasks: Task[]) {
+        const stack: Task[] = [];
+        for (const currTask of tasks) {
+            let prevTask: Task | undefined = undefined;
+            // looking for previous task with same or lower indentation
+            while (
+                stack.length > 0 &&
+                (prevTask = stack.last()) !== undefined &&
+                prevTask.indentation > currTask.indentation
```

Comment:
It looks like the indentation is done by counting characters...

What if the note has some tasks indented with spaces and some with tabs?

Reply:
not by counting, by raw string comparison. Please, see https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2765#issuecomment-2051601016 for further thoughts